thunder/dynamo/benchmark_utils.py

```diff
@@ -370,83 +404,127 @@ def _check_tensor(x):
     return any(tree_map(_check_tensor, report.example_input_meta))
 
 
-def check_timing(
+def check_memory_usage(m1, m2, m1_name, m2_name, test_name, rtol, atol):
+    m1_memory_str = get_pretty_memory_str(m1)
+    m2_memory_str = get_pretty_memory_str(m2)
+    if not check_threshold(m1, m2, rtol=rtol, atol=atol):
+        log_str = f"Max_allocated_memory for **{test_name}** requires investigation: {m1_name}({m1_memory_str}) and {m2_name}({m2_memory_str}) is not close (rtol={rtol}, atol={atol})\n"
+        return False, log_str
+    log_str = f"Max_allocated_memory: [{test_name}] {m1_name}({m1_memory_str}); {m2_name}({m2_memory_str})\n"
+    return True, log_str
+
+
+def check_metrics(
     folder_path,
     report,
     compile_fn1: Callable,
     compile_fn2: Callable,
     timer_fn: Callable,
-    timer_name: str,
-    rtol=0.5,
-    atol=0.0,
+    perf_rtol=0.5,
```

Comment:
Nit: maybe `time_rtol` instead of `perf_rtol`. The word "performance" is often used to describe both speed and memory usage, so it can be helpful to be more precise and use words like "speed" or "time"